### ISSUE TITLE
@map field name swap causes "Field is already defined on model" error

### ISSUE THREAD

**Issue by garth** [prisma/prisma#8687]
### Bug description

When using @map to rename fields, you get the following error when the model has a name that is mapped to a table that has a different field of the same name.

> Field "name" is already defined on model "Thing".

### How to reproduce

```
model Thing {
  name         String  @map("new_name")
  old_name  String  @map("name")
}
```

### Expected behavior

There should be no error

### Prisma information

```
model Thing {
  name         String  @map("new_name")
  old_name  String  @map("name")
}
```

### Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Node.js version: v16.6.1

### Prisma Version

```
prisma               : 2.29.0
@prisma/client       : 2.29.0
Current platform     : darwin
Query Engine         : query-engine 1be4cd60b89afa04b192acb1ef47758a39810f3a (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli 1be4cd60b89afa04b192acb1ef47758a39810f3a (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core 1be4cd60b89afa04b192acb1ef47758a39810f3a (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt 1be4cd60b89afa04b192acb1ef47758a39810f3a (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : 1be4cd60b89afa04b192acb1ef47758a39810f3a
Studio               : 0.419.0
```

**Comment by janpio** [comment]
I can confirm something going wrong here, weirdly my error message is different though:

```
model Thing {
  id       Int    @id @default(autoincrement())
  name     String @map("new_name")
  old_name String @map("name")
}
``` 

![image](https://user-images.githubusercontent.com/183673/129234499-3fc523f3-5357-4d3b-b9aa-fe52253966fc.png)

Note how it says`"Field "old_name" is already defined on model "Thing".`

Was that just a typo above @garth or do you indeed get a message about the field `name`?

**Comment by garth** [comment]
Sorry, I just typed out the schema roughly like I have in my project and renamed the fields, so probably got the fields the wrong way around. Glad that you could reproduce. Actually the bug has been there for quite some time, but I think 2.7 started failing the builds due to lint errors, before it was just an erroneous  error in vs code that could be safely ignored.

**Comment by pimeys** [comment]
I remember I fixed this already and that is a perfect excuse to do some `git bisect`. Our main branch allows your example, 2.29.0 does not. Bisect tells me I sure did fixed it in https://github.com/prisma/prisma-engines/pull/2347

Reading the pull request milestone, 3.4.0 is the first version that has this fixed, although I recommend updating to 3.7.0.
